chainer/variable.py

```diff
@@ -724,6 +724,16 @@ def __len__(self):
         """
         return len(self.array)
 
+    def __iter__(self):
+        if chainer.config.enable_backprop:
+            warnings.warn(
+                'Backprop through `Variable.__iter__` is slow. Consider'
+                ' using `chainer.functions.split_axis`.',
+                RuntimeWarning)
+        if self.ndim == 0:
+            raise TypeError('iteration over a 0-d array')
```

Comment:
How about do this check first, before emitting warning?

Reply:
Done.